javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -97,17 +98,11 @@ abstract class AbstractMap[K, V] protected () extends java.util.Map[K, V] { self
     throw new UnsupportedOperationException()
 
   def remove(key: Any): V = {
-    // scalastyle:off return
     val iter = entrySet.iterator
-    while (iter.hasNext) {
-      val item = iter.next()
-      if (key === item.getKey) {
-        iter.remove()
-        return item.getValue
-      }
+    iter.find(_.getKey === key).fold(null.asInstanceOf[V]) {
+      iter.remove()
```

Comment:
I rewrote the previous version with a `@tailrec` loop to avoid the `return` statement of the `while` loop.